configure.ac

```diff
@@ -151,6 +155,10 @@ AS_IF([test x"$hardening" != x"no"], [
        [HARDEN_CFLAGS="$HARDEN_CFLAGS -fPIE"
         HARDEN_LDFLAGS="$HARDEN_LDFLAGS -Wl,-pie"])])])
 
+  # If the current protobuf is newer than 4.22.x, add --copy-dt-needed-entries.
```

Comment:
Can you clarify why this is needed? It doesn't seem to be present before?

Reply:
aha, the protobuf package is providing and using abseil-cpp [1] as log/message process API. the abseil-cpp contains lots of .so libraries and they depends on each other, so add a --copy-dt-need-entries link flag to auto-link the symbols in those so files.

[1] https://github.com/protocolbuffers/protobuf/tree/22.x/third_party

Reply:
That doesn't sound right to me. If you're linking protobuf dynamically (you should be, in most distribution scenarios), then the NEEDED entries are evaluated transitively by ld. But if you're linking statically, then why are you depending on absl dynamically?

Reply:
> That doesn't sound right to me. If you're linking protobuf dynamically (you should be, in most distribution scenarios), then the NEEDED entries are evaluated transitively by ld. But if you're linking statically, then why are you depending on absl dynamically?

Yes, ld should auto find the needed dyanmic libraries to link, but i tested on a yocto distribution, the DSO error happens on ld, so I used this flag.

I sent a v2 commit, that using autotool to find the absl_* related libraries with pkg-config, it's better than using --copy-dt-need-entries 

v1->v2 changes: using pkg-config get the dynamic libraries instead of --copy-dt-need-entries
https://github.com/mobile-shell/mosh/compare/f9a8408e656644bcc679a0bea3de1ffc928deb4c..d9a1a6aac5a3b270449d09ec0d2a556807ab9287